hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/dto/AttributeSearchRequestJsonTest.kt

```diff
@@ -272,15 +283,15 @@ class AttributeSearchRequestJsonTest(@Autowired val objectMapper: ObjectMapper)
              "matchers": [
                {
                  "joinType": "AND",
-                 "stringMatchers": [
+                 "matchers": [
                    {
+                    "type": "String",
                      "attribute": "firstName",
                      "condition": "IS",
                      "searchTerm": "John"
-                   }
-                 ],
-                 "booleanMatchers": [
+                   },
                    {
+                     "type": "Boolean",
                      "attribute": "recall",
                      "condition": true
                    }
```

Comment:
With a single list of type matchers we don't have separate lists for each type which felt a bit clunky. Instead each matcher has a `type` discrimnator.